keep-runtime/src/workload.rs

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/// The error codes of workload execution.
+#[derive(Debug)]
+pub enum Error {
+    /// import module not found
+    ImportModuleNotFound(String),
+    /// import field not found
+    ImportFieldNotFound(String, String),
+    /// export not found
+    ExportNotFound,
+    /// call failed
+    CallFailed,
+    /// runtime error
+    RuntimeError,
+    /// I/O error
+    IoError(std::io::Error),
+}
+
+impl From<std::io::Error> for Error {
+    fn from(err: std::io::Error) -> Self {
+        Self::IoError(err)
+    }
+}
+
+/// Result type used throughout the library.
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Runs a WebAssembly workload.
+pub fn run<S: AsRef<[u8]>, T: std::borrow::Borrow<(S, S)>>(
+    bytes: impl AsRef<[u8]>,
+    args: impl IntoIterator<Item = S>,
+    envs: impl IntoIterator<Item = T>,
+) -> Result<Box<[wasmtime::Val]>> {
```

Comment:
I think we actually want this:

```suggestion
pub fn run<T: AsRef<[u8]>, U: AsRef<[u8]>, V: std::borrow::Borrow<(U, U)>>(
    bytes: impl AsRef<[u8]>,
    args: impl IntoIterator<Item = T>,
    envs: impl IntoIterator<Item = V>,
) -> Result<Box<[wasmtime::Val]>> {
```

In the previous code, the item in the iterator was forced to be the same for both `args` and `envs`. But they might actually be different. Hence, you need both `T: AsRef<[u8]>, U: AsRef<[u8]>`.